packages/button/src/Button.tsx

```diff
@@ -302,6 +306,8 @@ export default function Button(props: ButtonProps) {
       baseStyle,
       buttonSizes[size],
       buttonVariants[variant],
+      { [disabledStyle]: disabled },
+      { [disabledAnchor]: disabled && usesLinkElement(props) },
       className,
     ),
     disabled,
```

Comment:
We should remove `disabled` from `commonProps`, and apply it only to the button.

Reply:
Should we? It may be tricky when someone uses this with a custom root that supports `disabled`, a checkbox or radio button for example

Reply:
Maybe we could only exclude it from the link then, since we know it's an unsupported attribute.

Side-note: I was playing with the branch, and noticed that a recent fix for the button style in Firefox seems to have broken the styling of the button when rendered as an anchor tag. The text will show as being misaligned in that case.

Adding the following rules to the button styles seems to fix it:
```js
display: inline-flex;
align-items: center;
```

Reply:
Added this and did some cleanup on the styles as well, there were some duplication/overriding from the base to the variation/sizes
Also added a text decoration none when hovering, links would show underlined